### PR TITLE
Dbus api file scheme

### PR DESCRIPTION
**Pull request by wlozano0collabora**
Apply the new file name scheme to the implementation of D-BUS API.

### CODE REVIEW

**Review by justinjoy — APPROVED** [review]

File: agent/recorder-agent.c [hunk 1]
```diff
@@ -366,13 +366,22 @@ get_records (gchar * recording_dir, gchar * arg_edge_id, gchar * arg_record_id,
...
+    return edge_id;
+  }
+
+  file_list = g_array_new (TRUE, TRUE, sizeof (gchar *));
+
   while ((filename = g_dir_read_name (dir)) && !records_found) {
-    g_clear_pointer (&edge_id, g_free);
-    edge_id = strdup (filename);
+    g_clear_pointer (&edge_id_tmp, g_free);
+    edge_id_tmp = strdup (filename);
```
Comment (xhaakon) [inline agent/recorder-agent.c]: Use `g_strdup` for consistency.

File: agent/recorder-agent.c [hunk 2]
```diff
@@ -366,13 +366,22 @@ get_records (gchar * recording_dir, gchar * arg_edge_id, gchar * arg_record_id,
...
+
+    return edge_id;
+  }
+
+  file_list = g_array_new (TRUE, TRUE, sizeof (gchar *));
+
   while ((filename = g_dir_read_name (dir)) && !records_found) {
-    g_clear_pointer (&edge_id, g_free);
-    edge_id = strdup (filename);
+    g_clear_pointer (&edge_id_tmp, g_free);
```
Comment (xhaakon) [inline agent/recorder-agent.c]: If you make `edge_id_tmp` `g_autofree`, you don't need this line.

File: agent/recorder-agent.c [hunk 3]
```diff
@@ -387,43 +396,71 @@ get_records (gchar * recording_dir, gchar * arg_edge_id, gchar * arg_record_id,
...
+              g_strdup (file_id), file_start, file_end, st.st_size);
+      }
     }
   }
 
   if (dir)
     g_dir_close (dir);
   if (subdir)
     g_dir_close (subdir);
+  g_clear_pointer (&edge_id_tmp, g_free);
```
Comment (xhaakon) [inline agent/recorder-agent.c]: No need for this with if `edge_id_tmp` is `g_autofree`

File: agent/recorder-agent.c [hunk 4]
```diff
@@ -351,13 +347,17 @@ get_records (gchar * recording_dir, gchar * arg_edge_id, gchar * arg_record_id,
...
+  gchar *edge_id_tmp = NULL;
   gchar *edge_id = NULL;
   gchar *record_id = NULL;
   gchar *file_id = NULL;
   gchar *recording_edge_dir = NULL;
   gboolean records_found = FALSE;
   GStatBuf st;
   gchar *filename_full = NULL;
+  gint64 file_start = 0;
+  gint64 file_end = 0;
```
Comment (xhaakon) [inline agent/recorder-agent.c]: You  can limit the scope of these variables to the for loop that iterates `file_list`.

**Review by xhaakon — CHANGES_REQUESTED** [review]

File: agent/recorder-agent.c (hunk 1 above)
Comment (wlozano0collabora) [reply agent/recorder-agent.c]: Thanks for pointing this error.

File: agent/recorder-agent.c (hunk 2 above)
Comment (wlozano0collabora) [reply agent/recorder-agent.c]: Sorry, maybe I'm mistaken, but I though that ` g_autofree` will free `edge_id_tmp` after scope ending, but in this case, `edge_id_tmp` points to a string created with `g_strdup`, inside a loop, so will leak memory.

Comment (xhaakon) [reply agent/recorder-agent.c]: Ah, you're right. Reassignment won't free the original memory the pointer was referring to.

Comment (wlozano0collabora) [reply agent/recorder-agent.c]: However I think we can make this code clearer by moving some variable declarations inside the loop and adding `g_autofree`.

Comment (xhaakon) [reply agent/recorder-agent.c]: Yeah that's a good idea. Likely most of those variables that you call `g_clear_pointer()` on before assigning a value could better be declared inside the loop.

File: agent/recorder-agent.c (hunk 4 above)
Comment (wlozano0collabora) [reply agent/recorder-agent.c]: Thanks

**Review by xhaakon — APPROVED** [review]

**Pull request merged**